WebView2 JavaScript Reference Docs (#2315)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/webview2-api-reference.md

ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview

> ms.date: 10/06/2022

Reviewer: ```suggestion
ms.date: 02/02/2023
```
changing to this specific date should make merge conflict go away.  that's the date in "main" branch.  I wouldn't enter a later date until conflict resolves & build is clean.

* [Win32/C++](/microsoft-edge/webview2/reference/win32/index)

> * [JavaScript](/microsoft-edge/webview2/reference/javascript/index)

Reviewer 2: It would be helpful here to add some explanatory text to make it clear that this isn't just another language we make the webview2 APIs available in.

```suggestion
Additionally, WebView2 exposes a set of JavaScript APIs to the HTML documents running within WebView2 mostly to facilitate communication with the host application:

* [JavaScript](/microsoft-edge/webview2/reference/javascript/index)
```

* [Win32/C++](/microsoft-edge/webview2/reference/win32/index)

> Additionally, WebView2 exposes a set of JavaScript APIs to the HTML documents running within WebView2, mostly to facilitate communication with the host application:

Reviewer: @peiche-jessica - 
1. See the new tab-set format design at live site:  https://learn.microsoft.com/microsoft-edge/webview2/webview2-api-reference
2. Update your local "main" branch from server upstream main.
3. Merge main into this PR's branch.
4. Push commit.  
5. Add a 4th, rightmost tab if that layout makes sense to do.  So, add a 4th tab-heading like `##### [JavaScript](#tab/js)` just above the `---` line.  Keep blank lines above and below the tab-set terminator `---` line.

Author: I'd lean towards keeping "JavaScript APIs" separate from the tab set as they are meant to work together with the native WebView2 APIs (in the tab set).

Author: The rendered page with option 2

![image](https://user-images.githubusercontent.com/55165503/217645919-5be5936d-d3a1-4537-8998-611d793ebd79.png)

> Additionally, WebView2 exposes a set of JavaScript APIs to the HTML documents running within WebView2, mostly to facilitate communication with the host application:

Reviewer: ```suggestion
#### JavaScript

On the web side of a WebView2 app, WebView2 exposes a set of JavaScript APIs to the HTML documents running within WebView2, to facilitate communication with the host application:

* [JavaScript](/microsoft-edge/webview2/reference/javascript/)
```

> * [JavaScript](/microsoft-edge/webview2/reference/javascript/index)

Reviewer: This comment is to investigate whether to keep or remove "index" from end of path.  Seems like index might be ok per the View link in Doc Build table.

```suggestion
* [JavaScript](/microsoft-edge/webview2/reference/javascript/)
```

This line might end up being dup, depending on nearby review comment changes.

In main page of PR, in the bottom Doc Build report table, right-click the **View** link next to the reference/javascript/index.yml - paste the URL here, examine it: it is:

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/index?branch=pr-en-us-2315

Trim the start, probably keep index at end - I guess it's optional.
Trimmed result would almost certainly work:
/microsoft-edge/webview2/reference/javascript/index

Guessing that this would work too:
/microsoft-edge/webview2/reference/javascript/index
The way to test that omitting of "index" would be to try clicking this link that omits "index"

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/?branch=pr-en-us-2315

---

> <!-- ------------------------------ -->
> #### JavaScript
> On the web side of a WebView2 app, WebView2 exposes JavaScript APIs to the HTML documents that are running within the WebView2 control.  These JavaScript APIs are mostly to support communication with the host application.
> * [JavaScript](/microsoft-edge/webview2/reference/javascript/)

Reviewer: ```suggestion

<!-- ------------------------------ -->
#### JavaScript

On the web side of a WebView2 app, WebView2 exposes JavaScript APIs to the HTML documents that are running within the WebView2 control.  These JavaScript APIs are mostly to support communication with the host application.

* [JavaScript](/microsoft-edge/webview2/reference/javascript/)

```
adding blank lines per this repo's conventions

---

<!-- ------------------------------ -->
#### JavaScript

On the web side of a WebView2 app, WebView2 exposes JavaScript APIs to the HTML documents that are running within the WebView2 control.  These JavaScript APIs are mostly to support communication with the host application.

* [JavaScript](/microsoft-edge/webview2/reference/javascript/)

> 

Reviewer: ```suggestion
* [Web APIs](https://developer.mozilla.org/docs/Web/API) at MDN.
   * [Event](https://developer.mozilla.org/docs/Web/API/Event)
```
d/k if this fits here.
d/k if this link fits in See Also h2 below.
d/k if should link to specific API items or to the top level.

Author: It may add confusion if adding these 2 links. It should be okay not having them.
